### ISSUE TITLE
Player stranded in a separate world on first join

### ISSUE THREAD

**Issue by Wynnevir** [DRE2N/DungeonsXL#335]
We are trying to set up a test dungeon just to get an idea of the plugin.  And what I've found is that when a new map is created and saved, the first time someone uses /dxl play mapname, it takes them to an empty flatworld.
If you leave and redo the command, it loads the proper saved map, but having to enter every dungeon twice is terribly troublesome.
Any plan/idea of how to fix this bug?
MC Version 1.12.2(spigot)
Plugins (if this is relevant):
Skript
BetonQuest
Citizens
Bookshelves
Coreprotect
EssentialsX
Groupmanager
WorldEdit
WorldGuard
Multiverse
MV Portals
MCMMO
Dynmap
Viotifier
Perworld inventory
Holographic displays

thank you kindly!

**Comment by Tiplish** [comment]
Hi, juste to say that we have exactly the same bug on our server.

We alwas end the dungeon a first time before act the dungeon as playable.

If you made a change and play the dungeon immediatly, the change isn't saved. But in the second play time, it's ok.

**Comment by MarcatoSound** [comment]
Hey there,

My dev team and players on my server have discovered this actually seems to be a little random; sometimes players are thrust into a superflat world for no apparent reason, resulting in them having to leave and rejoin. It's not game-breaking, since they just have to type /dxl leave to get out, but it is a recurring problem worth mentioning.

**Comment by Sataniel98** [comment]
I guess your problem is, in this case, not caused by an actual mistake in the program.

If you create a map, it's always a flat world to begin with. In order to save your world from modifications caused by users playing the map and to make simultaneous playing possible, all maps are stored in the plugin folder (plugins/DungeonsXL/maps) and copied to the world container (the main directory) named "DXL_Game_{id}" when you play. So, the game instances of a map are always supposed to be in the state of the current map inside the "maps" folder, not in the state of an edit instance (a map called "DXL_Edit_{id}"). Edit world instances are not removed and stored in the maps folder instantly when a player leaves to prevent unnecessary world loading processes when the player decides to continue editing it shortly afterwards. Instead, this is done with a repeating task now and again and can be enforced with /dxl save (note that /dxl save also creates a backup if you have "backupMode" set to "ON_SAVE" or on "ON_DISABLE_AND_SAVE").

tl;dr; You are supposed to use /dxl save after editing a map.

However, since it seems to cause confusion, I think I'll do something to make it save automatically before the map is played or when an editor leaves.

**Comment by Wynnevir** [comment]
This problem is occurring **after** I use dxl save to save a map.
The steps I took:
1: /dxl create test
2: placed some blocks to make it clear you were in the test map.
3: /dxl save
4: /dxl leave
5:  Then /dxl play test.  It's at this point when you get moved to a flat world that is not the test map.
Doing dxl leave and then dxl play again will put me on the right map. I've done this a few times and everytime it's a dungeon I havn't used dxl play on before it first loads a blank world. leaving and running it a second time puts me in the right spot.

This has been spot on consistent for us thus far.
